rosidl_typesupport_connext_cpp/rosidl_typesupport_connext_cpp/__init__.py

```diff
@@ -110,11 +110,72 @@ def generate_dds_connext_cpp(
             # modify generated code to avoid unsed global variable warning
             # which can't be suppressed non-globally with gcc
             msg_filename = os.path.join(output_path, msg_name + '.h')
+            # TODO(karsten1987): Modify should take array of callbacks
+            # to avoid multiple file readings
             _modify(msg_filename, pkg_name, msg_name, _inject_unused_attribute)
 
+            plugin_filename = os.path.join(output_path, msg_name + 'Plugin.cxx')
+            _modify(plugin_filename, pkg_name, msg_name, _modify_plugin_serialize_function)
+
     return 0
 
 
+def _get_serialization_code(msg_name, indentation):
+    val = ("{{\n"
+           "{indentation}// MODIFIED FOR ROS2 PURPOSES\n"
+           "{indentation}const ConnextStaticMessageHandle * message_handle =\n"
+           "{indentation}  reinterpret_cast<const ConnextStaticMessageHandle *>(fake_sample);\n"
+           "{indentation}if (message_handle->raw_message) {{\n"
+           "{indentation}  memcpy(stream->_buffer, message_handle->raw_message, "
+           "*(message_handle->raw_message_length));\n"
+           "{indentation}  stream->_relativeBuffer = stream->_buffer;\n"
+           "{indentation}  stream->_tmpRelativeBuffer = stream->_buffer;\n"
+           "{indentation}  stream->_buffer = stream->_buffer;\n"
+           "{indentation}  //stream->_endian = \'\\x01\';\n"
+           "{indentation}  //stream->_nativeEndian = \'\\x01\';\n"
+           "{indentation}  //stream->_encapsulationKind = 1;\n"
+           "{indentation}  //stream->_zeroOnAlign = 0;\n"
+           "{indentation}  stream->_currentPosition = "
+           "stream->_buffer + *(message_handle->raw_message_length);\n"
+           "{indentation}  return RTI_TRUE;\n"
+           "{indentation}}}\n"
+           "{indentation}const {msg_name} * sample = reinterpret_cast<const {msg_name} *> "
+           "(message_handle->untyped_dds_message);\n"
+           .format(indentation=indentation, msg_name=msg_name))
+    return val
+
+
+def _modify_plugin_serialize_function(pkg_name, msg_name, lines):
+    # set include correctly - line 49 is the last generated include
```

Comment:
Since this seems pretty fragile can you add asserts to ensure that the line actually contains an include and that the next line doesn't.